### ISSUE TITLE
Crash when any script is closed

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#10359]
Godot 3.0 9575dbdf788e8a5154b3ec2f66913e731ac02850
Windows 10 64 bits

Godot crashes when I close a script, either with File->close menu or Ctrl+W.
Interestingly, it doesn't crash when it's a documentation file.

Error in the console right when it happens:
```
ERROR: RID_Owner<struct VisualServerCanvas::Item>::getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: 0
   At: d:\projets\info\godot\engine\godot_fork\core\rid.h:164
ERROR: VisualServerCanvas::canvas_item_set_visible: Condition ' !canvas_item ' is true.
   At: servers\visual\visual_server_canvas.cpp:331
```

![image](https://user-images.githubusercontent.com/1311555/29337224-df2b876a-8210-11e7-81f1-f7f318c83391.png)

**Comment by hpvb** [comment]
I can't reproduce this on Linux at 1be30f35a68a9fb2c156ded45c8689cc44f26f29 

Can you share the file(s) where this happens?

**Comment by Zylann** [comment]
Well... any?

**Comment by hpvb** [comment]
I can't reproduce it with a release_debug build or a regular build. On Linux, can you elaborate a bit more on exactly what you do? Sorry :-/

**Comment by Zylann** [comment]
I just close a script. With Ctrl+W, but it repros with File->Close too. 100% all the time.

**Comment by hpvb** [comment]
OK, I'll leave this to the Windows wranglers then. I'm sure someone will figure it out :)

**Comment by Zylann** [comment]
I found out the crash happens when I run the editor with a C++ debugger attached. Sounds like the crash is silent otherwise, plain luck?

**Comment by hpvb** [comment]
Can you paste the backtrace? I can probably fix it then.

**Comment by hpvb** [comment]
Uch, I didn't recognize it as a backtrace... I saw the color and thought it was part of the godot ui... *sigh* I'll have a look at it.

**Comment by reduz** [comment]
I cant reproduce this, but it seems to be windows only...
also, how do you use visual studio to debug? I would appreciate this info

**Comment by Zylann** [comment]
I just use QtCreator debugging, which is a wrapper around the Visual Debugger. Also the bug clearly shows access to a destroyed/corrupted object, so I wouldn't be surprised if that fails silently in other execution contexts.

**Comment by marcelofg55** [comment]
Is this crash still happening?

**Comment by Zylann** [comment]
It still happens.

The error in the console is the same:

```
ERROR: RID_Owner<struct VisualServerCanvas::Item>::getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: 0
   At: d:\projets\info\godot\engine\godot_fork\core\rid.h:164
ERROR: VisualServerCanvas::canvas_item_set_visible: Condition ' !canvas_item ' is true.
   At: servers\visual\visual_server_canvas.cpp:328
```

And I think the stack trace is similar (can't compare with old one, I still cannot see any image on Github thanks to Sky ISP):
![image](https://user-images.githubusercontent.com/1311555/31201609-8ebacf04-a95f-11e7-9a5b-645048a36b50.png)

**Comment by Zylann** [comment]
Still happens in alpha 2, though it doesn't produce a crash (see my comments above about the possible "why").
Same errors:
```
ERROR: RID_Owner<struct VisualServerCanvas::Item>::getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: 0
   At: c:\projects\godot-builds\godot\core\rid.h:164
ERROR: VisualServerCanvas::canvas_item_set_visible: Condition ' !canvas_item ' is true.
   At: servers\visual\visual_server_canvas.cpp:328
ERROR: RID_Owner<struct VisualServerCanvas::Item>::getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: 0
   At: c:\projects\godot-builds\godot\core\rid.h:164
ERROR: VisualServerCanvas::canvas_item_set_visible: Condition ' !canvas_item ' is true.
   At: servers\visual\visual_server_canvas.cpp:328
```

**Comment by akien-mga** [comment]
So IIUC it happens only when debugging Godot with MSVC?

**Comment by Zylann** [comment]
It happens everytime yeah, because the code accesses a bad pointer. It is less likely with a normal build because of chance.

**Comment by akien-mga** [comment]
I think you're the most capable for fixing it by now :D

**Comment by akien-mga** [comment]
Can you check if #15498 improved things?

**Comment by Zylann** [comment]
I tried on master and couldn't reproduce this in about 5 minutes. I guess the problem is gone.

**Comment by akien-mga** [comment]
Awesome :)

**Comment by akien-mga** [comment]
Fixed by #15498.
